src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java

```diff
@@ -323,9 +323,7 @@ private void visitReturn(DetailAST ast) {
          */
         public void visitLiteralReturn(int maxAssigned, Boolean voidReturn) {
             isVoidContext = voidReturn;
-            if (maxAllowed == null) {
```

Comment:
It doesn't matter if the if the variable is assigned or not, the instance only lasts for the current method and 1 method can always have 1 type of return, either by itself or with a value. For the same type of return, `maxAssigned` is always the same value.